stream.go

```diff
@@ -226,6 +234,19 @@ func (cs *clientStream) CloseSend() (err error) {
 	return
 }
 
+func (cs *clientStream) finish(err error) {
```

Comment:
The user could do write and read in 2 different goroutines.  i) Can cs.traceInfo.tr be accessed by multiple concurrent goroutines? ii) cs.traceInfo.tr = nil seems guarantee one of them will get a panic ...